src/components/modals/LogsModal.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import { Platform, View, ViewStyle } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import RNFS from 'react-native-fs'
+import Share, { ShareOptions } from 'react-native-share'
+
+import { saveLogOutput } from '../../actions/LogActions'
+import s from '../../locales/strings'
+import { useDispatch } from '../../types/reactRedux'
+import { WarningCard } from '../cards/WarningCard'
+import { CrossFade } from '../common/CrossFade'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow, ModalMessage, ModalTitle } from '../themed/ModalParts'
+import { OutlinedTextInput } from '../themed/OutlinedTextInput'
+import { ThemedModal } from '../themed/ThemedModal'
+interface LogsModalProps {
+  bridge: AirshipBridge<void>
+}
+
+export const LogsModal = ({ bridge }: LogsModalProps) => {
+  const theme = useTheme()
+  const dispatch = useDispatch()
+  const styles = getStyles(theme)
+  const [spinning, setSpinning] = React.useState(false)
+  const [userMessage, setUserMessage] = React.useState('')
+  const [safeToSend, setSafeToSend] = React.useState(true)
+  const dir = Platform.OS === 'android' ? RNFS.ExternalCachesDirectoryPath : RNFS.DocumentDirectoryPath
+
+  const handleShare = async () => {
+    const logs = await dispatch(saveLogOutput({ userMessage, send: false }))
+    const payload = JSON.stringify(logs, null, 2)
+    const username = logs.info.loggedInUser?.userName ?? ''
+    const path = `${dir}/edge-log-${username}.json`.replace('-.json', '.json')
+
+    const shareOptions: ShareOptions = {
+      title: s.strings.settings_button_export_logs,
+      subject: s.strings.settings_button_export_logs,
+      message: s.strings.settings_button_export_logs,
+      urls: [`file://${path}`],
+      type: 'application/json',
+      failOnCancel: false
+    }
+
+    await RNFS.writeFile(path, payload, 'utf8').catch(showError)
+    await Share.open(shareOptions).catch(showError)
+
+    bridge.resolve()
+  }
+
+  const handleSend = async () => {
+    setSpinning(true)
+    try {
+      await dispatch(saveLogOutput({ userMessage, send: true }))
+      bridge.resolve()
+    } catch (e: any) {
+      if (e.message.includes('⚠')) setSafeToSend(false)
+    }
```

Comment:
We should not be checking error messages for emojis. So yeah, we definitely need to do the safety checks in the modal itself.